controller/scheduler/fleet.py

```diff
@@ -112,6 +113,18 @@ def _create_container(self, name, image, command, unit, **kwargs):
             l.update({'cpu': '-c {}'.format(cpu)})
         else:
             l.update({'cpu': ''})
+        # set unit hostname
+        hostname = settings.UNIT_HOSTNAME
+        if hostname == "default":
+            l.update({'hostname': ''})
+        elif hostname == "application":
+            # replace underscore with dots, since underscore is not valid in DNS hostnames
+            dns_name = name.replace("_", ".")
```

Comment:
Maybe we should replace underscores with hyphens instead? I'm thinking dots typically imply subdomains, which may cause issues with some applications. Or maybe I'm entirely off-base.